### PR TITLE
Separator component

### LINKED ISSUE #9304: Add separator to model view dialogs
The model view dialogs need to have a `Separator` component or a way to manipulate the `css` classes to add separators after each `component`.  

Example use case - https://github.com/microsoft/azuredatastudio/pull/9303

### PR DESCRIPTION
**Pull request by aaomidi**
This PR fixes #9304

image

### CODE REVIEW

File: src/sql/base/browser/ui/separator/separator.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+
+export class Separator extends Disposable {
+	private readonly element: HTMLHRElement;
```
Comment (abist) [inline src/sql/base/browser/ui/separator/separator.ts]: `HTMLElement`

Comment (aaomidi) [reply src/sql/base/browser/ui/separator/separator.ts]: I think HRElement might be more useful here, we might need to change specific stuff related to HRElement.

Is there a reason you're suggesting us to do Element instead?

Comment (abist) [reply src/sql/base/browser/ui/separator/separator.ts]: Just being consistent with all other components. All other components use `HTMLElement`

Comment (aaomidi) [reply src/sql/base/browser/ui/separator/separator.ts]: Yeah I see what you mean but I would argue they should've also been scoped into the element they're creating.

Thanks!

**Pull request merged**